contracts/EverlongStrategy.sol

```diff
@@ -520,6 +546,183 @@ contract EverlongStrategy is BaseStrategy {
+        }
+
+        // Open the long. Return the maturity time and amount of bonds received.
+        (maturityTime, bondAmount) = IHyperdrive(hyperdrive).openLong(
+            asBase,
+            _toSpend,
+            _minOutput,
+            _minVaultSharePrice,
+            _extraData
+        );
+    }
+
+    /// @dev Preview the amount of bonds received from opening a position with
+    ///      the specified amount of strategy assets.
+    /// @param _toSpend Amount of strategy assets to spend.
+    /// @param _extraData Extra data to pass to hyperdrive.
+    /// @return bondAmount Amount of bonds that would be received.
+    function _previewOpenLong(
+        uint256 _toSpend,
+        bytes memory _extraData
+    ) internal view returns (uint256 bondAmount) {
+        // Call the preview function and return the expected amount of bonds
+        // to be received.
+        bondAmount = IHyperdrive(hyperdrive).previewOpenLong(
+            asBase,
+            _poolConfig,
+            _toSpend,
+            _extraData
+        );
+    }
+
+    /// @dev Preview the amount of assets received from closing the specified
+    ///      position.
+    /// @param _position Position to close.
+    /// @param _minOutput Minimum amount of proceeds to accept.
+    /// @param _extraData Extra data to pass to hyperdrive.
+    /// @return proceeds Amount of strategy assets that would be received.
+    function _closeLong(
+        IEverlongStrategy.EverlongPosition memory _position,
+        uint256 _minOutput,
+        bytes memory _extraData
+    ) internal returns (uint256 proceeds) {
+        // Close the long.
+        proceeds = IHyperdrive(hyperdrive).closeLong(
+            asBase,
+            _position,
+            _minOutput,
+            _extraData
+        );
+
+        // The proceeds must be wrapped if the strategy asset is a wrapped
+        // hyperdrive token.
+        if (isWrapped) {
+            // Approve the wrapped contract for the proceeds. Add one to the
+            // approval amount to save gas on future approvals by leaving the
+            // slot dirty.
+            ERC20(hyperdriveToken).forceApprove(address(asset), proceeds + 1);
+
+            // Wrap the proceeds.
+            proceeds = _wrap(convertToUnwrapped(proceeds));
+        }
+    }
+
+    /// @dev Preview the amount of assets received from closing the specified
+    ///      position.
+    /// @param _position Position to close.
+    /// @param _extraData Extra data to pass to hyperdrive.
+    /// @return proceeds Amount of strategy assets that would be received.
+    function _previewCloseLong(
+        IEverlongStrategy.EverlongPosition memory _position,
+        bytes memory _extraData
+    ) internal view returns (uint256 proceeds) {
+        // Call the preview function.
+        proceeds = IHyperdrive(hyperdrive).previewCloseLong(
+            asBase,
+            _poolConfig,
+            _position,
+            _extraData
+        );
+    }
```

Comment:
Same here. This seems like a refactor rather than something that has to change. I'm not making this suggestion with `_openLong` and `_closeLong` since it seems like those functions are reducing the complexity.

Reply:
totally on board